File: front-end/web/src/app/models/project-update-log.ts
import { Administrator } from "./administrator";

export class ProjectUpdateLog {
    id_project_update_log: number;
    description: string;
    author_id: number;
    administrator?: Administrator;
    content: string;

    constructor() {
        this.id_project_update_log = 0;
        this.description = '';
        this.author_id = 0;
        this.content = '';
        this.administrator = new Administrator;
    }
  }
  

File: back-end/models/ProjectUpdateLog.go
package models

import "time"

type ProjectUpdateLog struct {
	IDProjectUpdateLog uint          `gorm:"primaryKey" json:"id_project_update_log"`
	Description        string        `json:"description"`
	AuthorID           uint          `json:"author_id"`
	Author             Administrator `gorm:"foreignKey:AuthorID" json:"administrator"`
	Content            string        `json:"content"`
	CreatedAt          time.Time     `json:"created_at"`
	UpdatedAt          time.Time     `json:"updated_at"`
}
